k4FWCore/python/k4FWCore/utils.py

```diff
@@ -83,3 +85,43 @@ def load_file(opt_file: Union[TextIOWrapper, str, os.PathLike]) -> None:
     check_wrong_imports(str(code))
 
     exec(code, globals())
+
+
+class LoggingHandler(logging.StreamHandler):
+    def handleError(self, record):
+        # Re-raise SIGPIPE generated exception
+        excp = sys.exc_info()[1]  # from Python 3.11 can be replaced with sys.exception()
+        if isinstance(excp, BrokenPipeError):
+            raise excp
```

Comment:
```suggestion
        # from Python 3.11 can be replaced with sys.exception()
        if isinstance(excp := sys.exc_info()[1], BrokenPipeError):
            raise excp
```

Reply:
How far down do we have to go with the python support here? Maybe since we are touching this in any case we could also address the python 3.11 comment? Or are we not there yet?

Reply:
Outdated since https://github.com/key4hep/k4FWCore/pull/313